.github/workflows/code-generation.yaml

```diff
@@ -68,13 +68,58 @@ jobs:
     - name: Remove old code
       run: (! test -d ./astarteplatform) || rm -r ./astarteplatform
       working-directory: ./python
-    - name: Generate code
+    - name: Generate Python code
       run: python3 ./protoc.py "./../grpc/grpc_python_plugin"
       working-directory: ./python
-    - name: Commit the generated Python code
-      if: github.event_name == 'push'
+    - name: Upload Python code
+      uses: actions/upload-artifact@v3
+      with:
+        name: python
+        path: ./python
+  
+  rust_build:
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v4
+    - name: Install protoc
       run: |
-        git config user.name github-actions
-        git config user.email github-actions@github.com
-        git add ./python
-        git diff --staged --quiet || git commit -s -m "Generated Python code." && git push
+        curl -LO $PB_REL/download/v24.3/protoc-24.3-linux-x86_64.zip
+        unzip protoc-24.3-linux-x86_64.zip -d $HOME/.local
+        echo "$HOME/.local/bin" >> $GITHUB_PATH
+    - name: Install rust toolchain
+      uses: actions-rs/toolchain@v1
```

Comment:
**suggestion**: actions-rs is in slow maintenance mode

I would advise using `dtolnay/rust-toolchain` instead